src/main/java/classify/textfilecode/TextFileParser.java

```diff
@@ -0,0 +1,199 @@
+     * Returns true if filetype is .txt, false otherwise. 
+     * 
+     * @param textFileName Name of the file
+     * @return True or False depending on the filetype of file
+     */
+    public static Boolean textFileChecker(String textFileName){
+        String type = "";
+        int i = textFileName.lastIndexOf(DOT);
+        if (i >= 0) { 
+            type = textFileName.substring(i+1); 
+        }
+
+        return type.equals(TEXT_TYPE);
+    }
+
+    /**
+     * Prints the prompt for the user to select valid file
+     */
+    public static void promptForFileSelection(){
+        System.out.println(FILE_SELECTION_PROMPT);
+    }
+
+    /**
+     * Prints the prompt for the user to select valid file after entering a wrong input before.
+     */
+    public static void promptForFileSelectionAgain(){
+        System.out.println(REQUEST_TO_TRY_AGAIN);
+        System.out.println(FILE_SELECTION_PROMPT);
+    }
+    
+    public static void parseUserSelection(File currentDirectory, Scanner in, ArrayList <Student> masterStudentList){
+        File[] fileList = currentDirectory.listFiles();
+        String userInput = in.nextLine();
+
+        assert fileList != null;
+        int matchIndex = findMatchingFile(fileList, userInput);
+
+        if (matchIndex == -1){
+            System.out.println(NO_MATCH_FOUND);
+            promptForFileSelectionAgain();
+            parseUserSelection(TextFileHandler.CURRENT_DIRECTORY, in, masterStudentList);
+            return;
+        }
+        
+        parseTextFile(fileList, matchIndex, masterStudentList);
+                
+    }
+
+    private static void parseTextFile(File[] fileList, Integer matchIndex, ArrayList <Student> masterStudentList){
+        String fileNameString = fileList[matchIndex].getName().trim();
+        String fullFilePath = INPUT_TEXT_FILE_DIRECTORY + "/" + fileNameString;
+        try {
+            FileReader textfileReader = new FileReader(fullFilePath);
+            BufferedReader line = new BufferedReader(textfileReader);
+            UI.println("Fetching the data from " + fileNameString + ".");
+            fetchDataFromTextFile(line, masterStudentList);
+
+        } catch (FileNotFoundException e) {
+            UI.println(ERROR_ACCESSING_THE_FILE);
+        }
+
+    }
+
+    private static void fetchDataFromTextFile(BufferedReader line, ArrayList <Student> masterStudentList) {
+        try{
+            String firstLine = line.readLine();
+            String [] subjectInfoStrings = firstLine.split("Subject:");
+            String subjectName = subjectInfoStrings[1].trim();
+
+            while (line.ready()){
+                String input = line.readLine();
+                decodeTextFile(input, subjectName, masterStudentList);
+            }
+        } catch (IOException e) {
+            UI.println(ERROR_ACCESSING_THE_FILE);
+        }
+    }
+
+    private static void decodeTextFile (String textFileInput, String subjectName,
```

Comment:
A lot of repeated code here from the data package. Perhaps make a function for both for the purposes of reading in a text file?